### PR TITLE
docs: Add a demo

### PR DESCRIPTION
**Pull request by willser**

### CODE REVIEW

**Comment by Barugon** [comment]
Can you add a main function so this can be stand-alone?

File: README.md [hunk 1]
```diff
@@ -6,3 +6,51 @@
...
+    select_file: bool,
+
+    #[serde(skip)]
+    current: Option<PathBuf>,
+
+    opened_file: Option<PathBuf>,
+
+    #[serde(skip)]
+    open_file_dialog: Option<FileDialog>,
+}
```
Comment (Barugon) [inline README.md:32]: Serialization is not really needed here

File: README.md [hunk 2]
```diff
@@ -6,3 +6,51 @@
...
+            if (ui.button("Open")).clicked() {
+                let mut dialog = FileDialog::open_file(self.opened_file.clone());
+                dialog.open();
+                self.open_file_dialog = Some(dialog);
+            }
+
+            if let Some(dialog) = &mut self.open_file_dialog {
+                if dialog.show(&ctx).selected() {
+                    if let Some(file) = dialog.path() {
+                        self.opened_file = Some(file);
```
Comment (Barugon) [inline README.md:45]: Probably enough to simple do a println of the file

File: README.md [hunk 3]
```diff
@@ -6,3 +6,51 @@
...
+egui_file = "0.2.0"
+eframe = "0.19.0"
+
+````
+
+````rust
+
+#[derive(Deserialize, Serialize, Default)]
+pub struct Demo {
+    select_file: bool,
```
Comment (Barugon) [inline README.md:23]: Not used

File: README.md [hunk 4]
```diff
@@ -6,3 +6,51 @@
...
+
+    #[serde(skip)]
+    current: Option<PathBuf>,
```
Comment (Barugon) [inline README.md:26]: Not used

**Comment by Barugon** [comment]
I'm going to merge this and then modify it.

**Comment by willser** [comment]
Busy with work recently,sry.

**Pull request merged**